Source/Core/Common/MathUtil.h

```diff
@@ -30,6 +31,45 @@ constexpr auto Lerp(const T& x, const T& y, const F& a) -> decltype(x + (y - x)
   return x + (y - x) * a;
 }
 
+template <typename Dest, typename T>
+constexpr Dest SaturatingCast(T value)
+{
+  static_assert(std::is_integral<Dest>());
+
+  constexpr Dest lo = std::numeric_limits<Dest>::lowest();
+  constexpr Dest hi = std::numeric_limits<Dest>::max();
+
+  // T being a signed integer and Dest unsigned is a problematic case because the value will
+  // be converted into an unsigned integer, and u32(...) < 0 is always false.
+  if constexpr (std::is_integral<T>() && std::is_signed<T>() && std::is_unsigned<Dest>())
+  {
+    if (value < 0)
+      return lo;
```

Comment:
I feel like there should be a `static_assert(lo == 0)` or something like that here.

Reply:
Or perhaps replace `return lo;` with `return 0;`? What do you think would be clearer?

Reply:
I personally think the `static_assert` would be clearer, since it makes the `u32(...) < 0` assumption more explicit.

Reply:
Works for me